hastory/src/Hastory/Internal/Database.hs

```diff
@@ -0,0 +1,17 @@
+module Hastory.Internal.Database where
+
+import Control.Monad.IO.Unlift (MonadUnliftIO)
+import Database.Persist.Sqlite (IsSqlBackend)
+import Control.Monad.Trans.Reader (ReaderT)
+import Control.Monad.Logger (NoLoggingT)
+
+import qualified Data.Text as T
+import qualified Database.Persist.Sqlite as SQL
+
+runDb :: (MonadUnliftIO m, IsSqlBackend backend) => ReaderT backend (NoLoggingT (ResourceT m)) a -> m a
```

Comment:
Nit: You can use `SqlBackend` instead of `IsSqlBackend`, but this is fine.

Reply:
It turns out I *had* to use `SqlBackend` instead of `IsSqlBackend`, but I was wondering how you knew that? How did you interpret the types to know that that was necessary?